python/tests/unit/arcticdb/version_store/test_update.py

```diff
@@ -468,3 +469,76 @@ def test_update_pickled_data(lmdb_version_store):
         lmdb_version_store.update(symbol, df2)
+
+
+def test_update_non_sorted_exception(lmdb_version_store):
+    symbol = "bad_update"
+
+    num_initial_rows = 20
+    initial_timestamp = pd.Timestamp("2019-01-01")
+    dtidx = pd.date_range(initial_timestamp, periods=num_initial_rows)
+    df = pd.DataFrame({"c": [0 for i in range(0, num_initial_rows)]}, index=dtidx)
+    assert df.index.is_monotonic_increasing == True
+
+    lmdb_version_store.write(symbol, df)
+    info = lmdb_version_store.get_info(symbol)
+    assert info["sorted"] == "ASCENDING"
+
+    num_rows = 20
+    initial_timestamp = pd.Timestamp("2020-01-01")
+    dtidx = np.roll(pd.date_range(initial_timestamp, periods=num_rows), 3)
+    df2 = pd.DataFrame({"c": [0 for i in range(0, num_rows)]}, index=dtidx)
+    assert df2.index.is_monotonic_increasing == False
+
+    with pytest.raises(SortingException):
+        lmdb_version_store.update(symbol, df2)
+
+
+def test_update_non_sorted_multi_index_exception(lmdb_version_store):
+    symbol = "bad_write"
+    num_initial_rows = 20
+    num_rows = 20
+    initial_timestamp = pd.Timestamp("2020-01-01")
+    dtidx1 = pd.date_range(initial_timestamp, periods=num_initial_rows)
+    dtidx2 = np.arange(0, num_initial_rows)
+    dtidx = np.roll(pd.date_range(initial_timestamp, periods=num_initial_rows), 3)
+    df = pd.DataFrame(
+        {"c": [0 for i in range(0, num_rows)]},
+        index=pd.MultiIndex.from_arrays([dtidx1, dtidx2], names=["datetime", "level"]),
+    )
+    assert isinstance(df.index, MultiIndex) == True
+    assert df.index.is_monotonic_increasing == True
+    lmdb_version_store.write(symbol, df)
+
+    num_rows = 20
+    initial_timestamp = pd.Timestamp("2020-01-01")
+    dtidx1 = np.roll(pd.date_range(initial_timestamp, periods=num_initial_rows), 3)
+    dtidx2 = np.arange(0, num_initial_rows)
+    dtidx = np.roll(pd.date_range(initial_timestamp, periods=num_initial_rows), 3)
+    df = pd.DataFrame(
+        {"c": [0 for i in range(0, num_rows)]},
+        index=pd.MultiIndex.from_arrays([dtidx1, dtidx2], names=["datetime", "level"]),
+    )
+    assert isinstance(df.index, MultiIndex) == True
+    assert df.index.is_monotonic_increasing == False
+    lmdb_version_store.write(symbol, df)
+
+    with pytest.raises(SortingException):
+        lmdb_version_store.update(symbol, df)
+
+
+def test_update_non_sorted_range_index_exception(lmdb_version_store):
+    symbol = "bad_write"
+    num_rows = 20
+    dtidx = pd.RangeIndex(0, num_rows, 1)
+    df = pd.DataFrame({"c": [0 for i in range(0, num_rows)]}, index=dtidx)
+    assert df.index.is_monotonic_increasing == True
+    lmdb_version_store.write(symbol, df)
+
+    symbol = "bad_write"
+    num_rows = 20
+    dtidx = pd.RangeIndex(0, num_rows, 1)
+    df = pd.DataFrame({"c": [0 for i in range(0, num_rows)]}, index=dtidx)
+    assert df.index.is_monotonic_increasing == True
+    with pytest.raises(SortingException):
+        lmdb_version_store.update(symbol, df)
```

Comment:
Why is this a new test? `update` has always failed with non-timestamp indexed data hasn't it?

Reply:
yes it is correct. Matt asked for this test If I don't miss remember